From a055ecb91f2bfa414b625bebdc03b2d498e2ea0e Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Mon, 28 Apr 2014 10:58:21 +0800
Subject: [PATCH 3/3] anaconda: check metacity before run it

The metacity's schemas doesn't work, thus we would get this error:

GLib-GIO-ERROR **: Settings schema 'org.gnome.metacity' is not installed

We did have /usr/share/glib-2.0/schemas/org.gnome.metacity.gschema.xml,
we can run "glib-compile-schemas" to fix it, but the problem is that we
we fix it, progress bar will became ugly, so we will remove metacity
from anaconda's RDEPENDS and check whether /usr/bin/metacity before run
it to workaround the problem atm.

Upstream-Status: Inappropriate [WR-installer specific]

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 anaconda | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/anaconda b/anaconda
index 7715560..25e2086 100755
--- a/anaconda
+++ b/anaconda
@@ -42,6 +42,10 @@ def AnacondaShowWarning(message, category, filename, lineno, file=sys.stderr, li
     log.warning("%s" % warnings.formatwarning(message, category, filename, lineno, line))
 
 def startMetacityWM():
+    if not os.path.exists("/usr/bin/metacity"):
+        log.info("Skipping running /usr/bin/metacity since it doesn't exist.")
+        return 0
+
     childpid = os.fork()
     if not childpid:
         # after this point the method should never return (or throw an exception
@@ -49,7 +53,7 @@ def startMetacityWM():
         try:
             args = ['--display', ':1',
                     '--sm-disable']
-            iutil.execWithRedirect('metacity', args,
+            iutil.execWithRedirect('/usr/bin/metacity', args,
                                    stdout='/dev/null', stderr='/dev/null')
         except BaseException as e:
             # catch all possible exceptions
-- 
1.8.3.1

